packages/ui-text/src/Text/styles.js

```diff
@@ -0,0 +1,165 @@
+ */
+const generateStyle = (theme, themeOverride, props, state) => {
+  // get the theme variables object for the component
+  const componentTheme = generateComponentTheme(theme, themeOverride)
+  const {
+    size,
+    wrap,
+    weight,
+    fontStyle,
+    transform,
+    lineHeight,
+    letterSpacing,
+    color
+  } = props
+
+  const colorVariants = {
+    primary: componentTheme.primaryColor,
+    secondary: componentTheme.secondaryColor,
+    'primary-inverse': componentTheme.primaryInverseColor,
+    'secondary-inverse': componentTheme.secondaryInverseColor,
+    success: componentTheme.successColor,
+    brand: componentTheme.brandColor,
+    warning: componentTheme.warningColor,
+    danger: componentTheme.dangerColor,
+    alert: componentTheme.alertColor
+  }
+
+  const wrapStyle =
+    wrap === 'break-word'
+      ? {
+          overflowWrap: 'break-word',
+          wordBreak: 'break-word',
+          hyphens: 'auto'
+        }
+      : {}
+
+  const weightStyle = {
+    normal: componentTheme.fontWeightNormal,
+    light: componentTheme.fontWeightLight,
+    bold: componentTheme.fontWeightBold
+  }
+
+  const fontSizeVariants = {
+    'x-small': componentTheme.fontSizeXSmall,
+    small: componentTheme.fontSizeSmall,
+    medium: componentTheme.fontSizeMedium,
+    large: componentTheme.fontSizeLarge,
+    'x-large': componentTheme.fontSizeXLarge,
+    'xx-large': componentTheme.fontSizeXXLarge
+  }
+
+  const lineHeightVariants = {
+    default: componentTheme.lineHeight,
+    fit: componentTheme.lineHeightFit,
+    condensed: componentTheme.lineHeightCondensed,
+    double: componentTheme.lineHeightDouble
+  }
+
+  const letterSpacingVariants = {
+    normal: componentTheme.letterSpacingNormal,
+    condensed: componentTheme.letterSpacingCondensed,
+    expanded: componentTheme.letterSpacingExpanded
+  }
+
+  const textOrInputStyle = {
+    '&:focus': {
+      outline: 'none'
+    },
+    color: colorVariants[color],
+    ...wrapStyle,
+    fontWeight: weightStyle[weight],
+    fontStyle: fontStyle,
+    fontSize: fontSizeVariants[size],
+    lineHeight: lineHeightVariants[lineHeight],
+    letterSpacing: letterSpacingVariants[letterSpacing],
+    textTransform: transform
+  }
+
+  return {
+    text: {
+      label: 'text',
+      fontSize: componentTheme.fontFamily,
+      'sub, sup': {
+        fontSize: '75%',
+        lineHeight: 0,
+        position: 'relative',
+        verticalAlign: 'baseline'
+      },
+      sup: {
+        top: '-0.4em'
+      },
+      sub: {
+        bottom: '-0.4em'
+      },
+      'pre, code': {
+        all: 'initial',
+        fontFamily: componentTheme.fontFamilyMonospace
+      },
+      'i, em': {
+        fontStyle: 'italic'
+      },
+      'b, strong': {
+        fontWeight: componentTheme.fontWeightBold
+      },
+      p: {
+        display: 'block',
+        padding: 0,
+        margin: componentTheme.paragraphMargin
+      },
+      '& input': {
```

Comment:
This is needed for InPlaceEdit, please use styling like this: https://github.com/instructure/instructure-ui/commit/5968508755adf2c9c5c33bc319a3abc522a87866
(You can preview it in InPlaceEdit component, when you click to edit the text and it becomes an input)

Reply:
done